Sources/SwifterSwift/SwiftStdlib/SequenceExtensions.swift

```diff
@@ -161,110 +178,120 @@ public extension Sequence {
     ///
-    /// - Parameter keyPath: Key path of the `AdditiveArithmetic` property.
-    /// - Returns: The sum of the `AdditiveArithmetic` properties at `keyPath`.
-    func sum<T: AdditiveArithmetic>(for keyPath: KeyPath<Element, T>) -> T {
+    /// - Parameter map: Getter for  the `AdditiveArithmetic` property.
+    /// - Returns: The sum of the `AdditiveArithmetic` properties at `map`.
+    func sum<T: AdditiveArithmetic>(for map: (Element) -> T) -> T {
         // Inspired by: https://swiftbysundell.com/articles/reducers-in-swift/
-        return reduce(.zero) { $0 + $1[keyPath: keyPath] }
+        return reduce(.zero) { $0 + map($1) }
+    }
+    
+    /// SwifterSwift: Product of all elements in array.
+    ///
+    ///        ["James", "Wade", "Bryant"].product(for: \.count) -> 120
+    ///
+    /// - Parameter keyPath: Key path of the `Numeric` property.
+    /// - Returns: product of the array's elements.
+    func product<T: Numeric>(for map: (Element) -> T) -> T {
+        reduce(1) { $0 * map($1) }
     }
 
-    /// SwifterSwift: Returns the first element of the sequence with having property by given key path equals to given
+    /// SwifterSwift: Returns the first element of the sequence with having the value by given map function equals to given
     /// `value`.
     ///
     /// - Parameters:
-    ///   - keyPath: The `KeyPath` of property for `Element` to compare.
+    ///   - map: Function for `Element` to compare.
     ///   - value: The value to compare with `Element` property.
-    /// - Returns: The first element of the collection that has property by given key path equals to given `value` or
+    /// - Returns: The first element of the collection that has property by given map function equals to given `value` or
     /// `nil` if there is no such element.
-    func first<T: Equatable>(where keyPath: KeyPath<Element, T>, equals value: T) -> Element? {
-        return first { $0[keyPath: keyPath] == value }
+    func first<T: Equatable>(where map: (Element) throws -> T, equals value: T) rethrows -> Element? {
+        return try first { try map($0) == value }
     }
 }
 
 public extension Sequence where Element: Equatable {
-    /// SwifterSwift: Check if array contains an array of elements.
+    /// SwifterSwift: Check if sequence contains elements of another sequence.
     ///
     ///        [1, 2, 3, 4, 5].contains([1, 2]) -> true
     ///        [1.2, 2.3, 4.5, 3.4, 4.5].contains([2, 6]) -> false
     ///        ["h", "e", "l", "l", "o"].contains(["l", "o"]) -> true
     ///
-    /// - Parameter elements: array of elements to check.
+    /// - Parameter elements: sequence of elements to check.
     /// - Returns: true if array contains all given items.
     /// - Complexity: _O(m·n)_, where _m_ is the length of `elements` and _n_ is the length of this sequence.
-    func contains(_ elements: [Element]) -> Bool {
+    func contains<T: Sequence>(_ elements: T) -> Bool where Element == T.Element {
```

Comment:
Although small, this could be a source breaking change, as `[Element]` is more specialized, in a place where this is being choose by overload resolution, with this change other method can be considered ambiguous or be choose instead. Let's keep it as is, and if we would remove, let's do in a separate PR where we deprecate the old one and then introduce the new generic signature.

Reply:
Alright, I'll work on that as soon as I can.